NitroxServer/Serialization/ServerProtoBufSerializer.cs

```diff
@@ -1,132 +1,119 @@
-                    }
+                    Model.Add(type, true);
+
+                    ManuallyRegisterUweProtoMembers(type.GetMembers(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance), type);
                 }
             }
-
-            if (ignoredTypeErrors.Count > 0)
+            catch (Exception ex)
             {
-                Log.Debug($"[ServerProtoBufSerializer] Has thrown an error registering: {string.Join(", ", ignoredTypeErrors)} from {assemblyName}. However it's probably caused from different Newtonsoft.Json versions of Oculus and Nitrox and can be ignored.");
+                Log.Error(ex, $"ServerProtoBufSerializer has thrown an error registering the type: {type} from {assemblyName}");
             }
         }
+    }
 
-        private bool HasUweProtoContract(Type type)
+    private static bool HasUweProtoContract(Type type)
+    {
+        foreach (object o in type.GetCustomAttributes(false))
         {
-            foreach (object o in type.GetCustomAttributes(false))
+            if (o.GetType().ToString().Contains("ProtoContractAttribute"))
             {
-                if (o.GetType().ToString().Contains("ProtoContractAttribute"))
-                {
-                    return true;
-                }
+                return true;
             }
-
-            return false;
         }
 
-        private void ManuallyRegisterUweProtoMembers(MemberInfo[] info, Type type)
+        return false;
+    }
+
+    private void ManuallyRegisterUweProtoMembers(MemberInfo[] info, Type type)
+    {
+        foreach (MemberInfo property in info)
         {
-            foreach (MemberInfo property in info)
+            if (property.DeclaringType != type)
+            {
+                continue;
+            }
+
+            foreach (object customAttribute in property.GetCustomAttributes(false))
             {
-                if (!(property.DeclaringType != type))
+                Type attributeType = customAttribute.GetType();
+                if (attributeType.ToString().Contains("ProtoMemberAttribute"))
                 {
-                    foreach (object customAttribute in property.GetCustomAttributes(false))
-                    {
-                        Type attributeType = customAttribute.GetType();
-                        if (attributeType.ToString().Contains("ProtoMemberAttribute"))
-                        {
-                            int tag = (int)attributeType.GetProperty("Tag", BindingFlags.Public | BindingFlags.Instance).GetValue(customAttribute, emptyArray);
-                            Model[type].Add(tag, property.Name);
-                        }
-                    }
+                    int tag = (int)attributeType.GetProperty("Tag", BindingFlags.Public | BindingFlags.Instance).GetValue(customAttribute, emptyArray);
+                    Model[type].Add(tag, property.Name);
```

Comment:
We can use `is ProtoMemberAttribute protoAttr` to get the tag now, since DLLs are publicized.